src/NexusMods.DataModel/Synchronizer/SynchronizerState.cs

```diff
@@ -1,14 +1,21 @@
 using System.Reactive.Disposables;
 using DynamicData.Kernel;
 using NexusMods.Abstractions.Loadouts.Ids;
+using ReactiveUI;
 using ReactiveUI.Fody.Helpers;
 
 namespace NexusMods.DataModel.Synchronizer;
 
-public class SynchronizerState
+public class SynchronizerState : ReactiveObject
 {
-    [Reactive] public bool Busy { get; set; } = false;
-    
+    private volatile bool _busy = false;
+
+    public bool Busy
+    {
+        get => _busy;
+        set => this.RaiseAndSetIfChanged(ref _busy, value);
```

Comment:
Why did this swap to ReactiveObject?

Reply:
@Al12rs thought he had a threading issue while debugging last Thursday, so he wanted to check if making the structure volatile would have helped them.

In practice `volatile` is generally only used to:

1. Erect a memory barrier (cannot reorder operations before/after)
2. Ensure value is immediately written back to in loops

In any case, I've specified last week that `volatile` keyword here is redundant given how the variable is set and read.

I imagine the `ReactiveObject` comes from the fact they wanted to call `RaiseAndSetIfChanged` manually here since they made a private field which was volatile.

Reply:
Right, this went through a number of iterations. Originally the Busy property was marked with the `[Reactive]` attribute without inheriting from `ReactiveObject`, which made it not notify of changes correctly.
`[Reactive]` requires in some way inheriting from `ReactiveObject`.

Due to the the value being set in a critical section, the reactive weaver wouldn't work even with `ReactiveObject`, I tried a few things and the working solution was to call `RaiseAndSetIfChanged` with a backing field.
`RaiseAndSetIfChanged` is a method from `ReactiveObject` so still inheriting from that.

I tested removing volatile, which was one of the experiments and it seems to work fine, so I'll remove that.